tests/integration/components/bs-modal-test.js

```diff
@@ -502,3 +502,44 @@ test('Renders in place (no wormhole) if renderInPlace is set', function(assert)
   assert.equal(this.$('.modal').length, 1, 'Modal exists.');
   assert.notEqual(this.$('.modal').parent().attr('id'), 'ember-bootstrap-modal-container');
 });
+
+test('Removes "modal-open" class when component is removed from view', function(assert) {
+  this.set('renderComponent', true);
+  this.render(hbs`{{#if renderComponent}}{{#bs-modal title="Simple Dialog"}}Hello world!{{/bs-modal}}{{/if}}<div id="ember-bootstrap-modal-container"></div>`);
+
+  let done = assert.async();
+
+  // wait for fade animation
+  setTimeout(() => {
+    assert.ok($('body').hasClass('modal-open'), 'body element has "modal-open" class.');
+
+    this.set('renderComponent', false);
+
+    // wait for fade animation
+    setTimeout(() => {
```

Comment:
The second `setTimeout` in both tests should be omitted. Make the assertion directly after setting `renderComponent` to false. 

We do not have a problem when the modal is closed programmatically, by setting `open` to false, in which case the fade out transition is made. Only when it is "hard" removed as you do it within your template. In this case the fade out transition is never started, so let's test the correct cleanup immediately!

Apart from that this looks good! Thanks!!!